packages/init/src/steps/sendStarterData.ts

```diff
@@ -0,0 +1,221 @@
+      headers: {
+        Authorization: authorization,
+        repository,
+      },
+    })
+    .then((res) => {
+      return Array.isArray(res.data) ? res.data.map((model) => model.id) : [];
+    });
+
+  if (remoteSlices.length) {
+    // do prompt about slices
+
+    const pushAnyway = await inquirer
+      .prompt<{ pushSlices: boolean }>([
+        {
+          type: "confirm",
+          name: "pushSlices",
+          default: false,
+          message:
+            "Your repository already contains Slices. Do you want to continue pushing your local Slices?",
+        },
+      ])
+      .then((res) => res.pushSlices);
+
+    if (pushAnyway === false) return Promise.resolve(true);
+  }
+
+  const spinner = logs.spinner("Pushing existing models to your repository");
+  spinner.start();
+
+  if (smJson.content && smJson.content.libraries) {
+    const libs = Libraries.libraries(cwd, smJson.content.libraries);
+
+    const components = libs.reduce<Array<Component>>((acc, lib) => {
+      return [...acc, ...lib.components];
+    }, []);
+
+    const sendToS3 = await setupS3(
+      repository,
+      authorization,
+      endpoints.AclProvider
+    );
+
+    const modelPromises = components.map(async ({ screenshotPaths, model }) => {
+      const variationsReq = model.variations.map(async (variation) => {
+        const pathToScreenShot = screenshotPaths[variation.id];
+        if (pathToScreenShot && pathToScreenShot.path) {
+          const imageUrl = await sendToS3(
+            model.id,
+            variation.id,
+            pathToScreenShot.path
+          ).catch(() => "");
+          return {
+            ...variation,
+            ...(imageUrl ? { imageUrl } : {}), // is image url needed by the api?
+          };
+        } else {
+          return variation;
+        }
+      });
+      const variations = await Promise.all(variationsReq);
+      return {
+        ...model,
+        variations,
+      };
+    });
+
+    const models = await Promise.all(modelPromises);
+
+    const p = models.map((model) => {
+      const data = Slices.fromSM(model);
+      const updateOrInsertUrl = `${endpoints.Models}slices/${
+        remoteSlices.includes(model.id) ? "update" : "insert"
+      }`;
+      return axios
+        .post(updateOrInsertUrl, data, {
+          headers: {
+            Authorization: authorization,
+            repository,
+          },
+        })
+        .then((res) => {
+          if (Math.floor(res.status / 100) !== 2) {
+            console.log(
+              `[ERROR: ${res.status}] SENDING SLICE ${model.id} | ${res.statusText}`
+            );
+          }
+        })
+        .catch((err) => {
+          if (err instanceof Error) {
+            console.log(`[ERROR] SENDING SLICE ${err.message}`);
+          } else {
+            console.log(`[ERROR] SENDING SLICE ${String(err)}`);
+          }
+        });
+    });
+
+    await Promise.all(p);
+  }
+
+  spinner.succeed();
```

Comment:
We don't stop the spinner on error case.
We also don't stop the APP if the push fails, is it normal ?